registry/client/transport/http_reader.go

```diff
@@ -19,24 +19,23 @@ var (
 )
 
 // ReadSeekCloser combines io.ReadSeeker with io.Closer.
-type ReadSeekCloser interface {
-	io.ReadSeeker
-	io.Closer
-}
+//
+// Deprecated: use [io.ReadSeekCloser].
+type ReadSeekCloser = io.ReadSeekCloser
```

Comment:
- It looks like this is backporting https://github.com/distribution/distribution/pull/3788 - was that needed to make it work with go1.18, or accidentally included in this commit? (if it's not needed for the go1.18 update, it's better to keep it in a separate PR)

Reply:
Yes, it makes it more visible that it was back ported; and it allows you to leave a comment on the PR _why_ it needs to be backported

Reply:
I will come up with another PR to back port it, just keep this one as an upgrade to golang.